spectre-meltdown-checker.sh

```diff
@@ -345,63 +345,58 @@ is_cpu_vulnerable()
-					[ -z "$variant2" ] && variant2=immune
-				fi
-
-				# for variant3, only A75 is vulnerable
-				if [ "$cpuarch" = 8 ] && [ "$cpupart" = 0xd0a ]; then
-					_debug "checking cpu$i: arm A75 vulnerable to meltdown"
+					[ -z "$variant3" ] && variant3=immune
 					variant3=vuln
-				else
-					_debug "checking cpu$i: this arm non vulnerable to meltdown"
+					[ -z "$variant4" ] && variant4=immune
+					_debug "checking cpu$i: this armv7 non vulnerable to variants 3 & 4"
+				elif [ "$cpuarch" = 8 ] && echo "$cpupart" | grep -Eq '^0x(d07|d08)$'; then
+					variant1=vuln
+					variant2=vuln
+					variant3=vuln
+					[ -z "$variant3a" ] && variant3a=immune
+					[ -z "$variant4" ] && variant4=immune
+					_debug "checking cpu$i: this armv7 non vulnerable to variants 3a & 4"
+				elif [ "$cpuarch" = 8 ] && echo "$cpupart" | grep -Eq '^0x(d09|d0b)$'; then
+					variant1=vuln
+					variant2=vuln
 					[ -z "$variant3" ] && variant3=immune
-				fi
-
-				# for variant3a, only A15/A57/A72 are vulnerable
-				if [ "$cpuarch" = 8 ] && echo "$cpupart" | grep -Eq '^0x(c0f|d07|d0a)$'; then
-					_debug "checking cpu$i: arm A15-A57-A72 vulnerable to variant3a"
-					variant3a=vuln
-				else
-					_debug "checking cpu$i: this arm non vulnerable to variant3a"
 					[ -z "$variant3a" ] && variant3a=immune
-				fi
-
-				# for variant4, only A57-72-73-75 are vulnerable
-				if [ "$cpuarch" = 8 ] && echo "$cpupart" | grep -Eq '^0xd0[789a]$'; then
-					_debug "checking cpu$i: arm A57-A72-A73-A75 vulnerable to variant4"
 					variant4=vuln
+					_debug "checking cpu$i: this armv8 non vulnerable to variants 3 & 3a"
+				elif [ "$cpuarch" = 8 ] && echo "$cpupart" | grep -Eq '^0x(c0a)$'; then
+					variant1=vuln
+					variant2=vuln
+					variant3=vuln
+					[ -z "$variant3a" ] && variant3a=immune
+					variant4=vuln
+					_debug "checking cpu$i: this armv8 non vulnerable to variant 3a"
 				else
-					_debug "checking cpu$i: this arm non vulnerable to variant4"
-					[ -z "$variant4" ] && variant4=immune
+					variant1=vuln
+					variant2=vuln
+					variant3=vuln
+					variant3a=vuln
+					variant4=vuln
+					_debug "checking cpu$i: this arm unknown"
 				fi
 			fi
 			_debug "is_cpu_vulnerable: for cpu$i and so far, we have <$variant1> <$variant2> <$variant3> <$variant3a> <$variant4>"
 		done
 	fi
-	# from the information we have for now, it seems that CPUs that are vulnerable to variant1 are also vulnerable to variant4
-	[ -z "$variant4" ] && variant4=$variant1
-	_debug "is_cpu_vulnerable: temp results are <$variant1> <$variant2> <$variant3> <$variant3a> <$variant4>"
-	# if at least one of the cpu is vulnerable, then the system is vulnerable
-	[ "$variant1"  = "immune" ] && variant1=1  || variant1=0
-	[ "$variant2"  = "immune" ] && variant2=1  || variant2=0
-	[ "$variant3"  = "immune" ] && variant3=1  || variant3=0
-	[ "$variant3a" = "immune" ] && variant3a=1 || variant3a=0
-	[ "$variant4"  = "immune" ] && variant4=1  || variant4=0
```

Comment:
Thanks for the updates regarding the ARM models!

However lines 398-404: why did you delete them? Their job is to convert the temporary values "immune" or "vuln" or "" into 0 or 1 that is used and expected in the remaining of the script (those lines are executed for all archs, not only ARM).

Reply:
I've just unbroken that bit.